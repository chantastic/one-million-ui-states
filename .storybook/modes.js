/*
Lesson 13: Snapshot multiple viewports in Chromatic with modes

Chromatic modes let you to generate snapshots for a singlne state/story.

Uncomment (or write) the code below to define modes.
*/

// export const viewportModes = {
//   small: { name: "Small", viewport: { width: "640px" } },
//   medium: { name: "Medium", viewport: { width: "768px" } },
//   large: { name: "Large", viewport: { width: "1024px" } },
// };
