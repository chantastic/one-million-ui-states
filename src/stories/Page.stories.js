import { within, userEvent, expect } from "@storybook/test";
import { viewportModes } from "../../.storybook/modes";

import { Page } from "./Page";

export default {
  title: "Example/Page",
  component: Page,
  parameters: {
    layout: "fullscreen",
    chromatic: {
      modes: {
        mobile: viewportModes["small"],
        desktop: viewportModes["large"],
      },
    },
  },
};

export const LoggedOut = {};

/*
Lesson 5: Anatomy of a play function

Play functions use testing-library functions to create stories from user interactions.
They can include user events and Jest-style assertions.

You can observe the interactions that created a story in the Interactions Pane of the Addon Panel.

- Find and remove the click event that logs in the user
- What happens to both the story and interactions pane? 
*/

export const LoggedIn = {
  play: async ({ canvasElement }) => {
    const canvas = within(canvasElement);
    const loginButton = canvas.getByRole("button", { name: /Log in/i });
    await expect(loginButton).toBeInTheDocument();
    await userEvent.click(loginButton);
    await expect(loginButton).not.toBeInTheDocument();

    const logoutButton = canvas.getByRole("button", { name: /Log out/i });
    await expect(logoutButton).toBeInTheDocument();
  },
};

/*
Lesson 6: Compose play functions

Play functions can also be composed from other stories.
Call play functions from their stories.

Uncomment (or rewrite) the story below.
Note that it calls the play function from the LoggedIn story.
*/

// export const LogInThenLogOut = {
//   play: async (context) => {
//     await LoggedIn.play(context);

//     const canvas = within(context.canvasElement);
//     const loginButton = await canvas.getByRole("button", { name: /Log out/i });
//     await userEvent.click(loginButton);

//     await expect(canvas.getByText(/Log in/));
//   },
// };
