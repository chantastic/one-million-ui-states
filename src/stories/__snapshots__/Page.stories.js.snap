// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example/Page LogInThenLogOut play-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log in",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "focused": true,
      "name": "Sign up",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "level": 2,
      "name": "Pages in Storybook",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "We recommend building UIs with a",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "component-driven",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "process starting with atomic components and ending with pages.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Assemble data in the page component from your services. You can mock these services out using Storybook.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Tip",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " Adjust the width of the canvas with the",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Viewports addon in the toolbar",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "example-page--log-in-then-log-out",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Example/Page LoggedIn play-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Welcome, ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Jane Doe",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "!",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log out",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "level": 2,
      "name": "Pages in Storybook",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "We recommend building UIs with a",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "component-driven",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "process starting with atomic components and ending with pages.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Assemble data in the page component from your services. You can mock these services out using Storybook.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Tip",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " Adjust the width of the canvas with the",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Viewports addon in the toolbar",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "example-page--logged-in",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;

exports[`Example/Page LoggedOut smoke-test 1`] = `
{
  "checked": undefined,
  "children": [
    {
      "checked": undefined,
      "children": undefined,
      "level": 1,
      "name": "Acme",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Log in",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Sign up",
      "pressed": undefined,
      "role": "button",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "level": 2,
      "name": "Pages in Storybook",
      "pressed": undefined,
      "role": "heading",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "We recommend building UIs with a",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "component-driven",
      "pressed": undefined,
      "role": "link",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " ",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "process starting with atomic components and ending with pages.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Render pages with mock data. This makes it easy to build and review page states without needing to navigate to them in your app. Here are some handy patterns for managing page data in Storybook:",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Use a higher-level connected component. Storybook helps you compose such data from the "args" of child component stories",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Assemble data in the page component from your services. You can mock these services out using Storybook.",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Tip",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": " Adjust the width of the canvas with the",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
    {
      "checked": undefined,
      "children": undefined,
      "name": "Viewports addon in the toolbar",
      "pressed": undefined,
      "role": "text",
      "value": undefined,
    },
  ],
  "name": "example-page--logged-out",
  "pressed": undefined,
  "role": "WebArea",
  "value": undefined,
}
`;
